### ISSUE TITLE
[script issue] - newly created checkpoint already contain a file.

### ISSUE THREAD

**Issue by raj-ritu17** [intel-analytics/ipex-llm#11099]
following the  below guidelines : 
https://github.com/intel-analytics/ipex-llm/tree/main/python/llm/example/GPU/LLM-Finetuning/QLoRA/alpaca-qlora 

using --> qlora_finetune_llama2_7b_arc_2_card.sh

issue: 
main issue is, whenever it create a new checkpoint, it also create a file inside it named 'rng_state_1.pth' and when tmp-checkpoint-* trying to rename the current folder it can't replace as its not empty. 

```
OSError: [Errno 39] Directory not empty: './ipex-llm-qlora-alp
aca/tmp-checkpoint-700' -> './ipex-llm-qlora-alpaca/checkpoint-700'
```

ss for eg:
![image](https://github.com/intel-analytics/ipex-llm/assets/110594625/b64bd635-2fe2-4f8f-8313-29d9ed5166be)

complete error:
` 15%|█▌        | 700/4659 [14:14<4:33:24,  4.14s//home/intel/miniconda3/envs/ft_Qlora/lib/python3.11/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
Traceback (most recent call last):
  File "/home/intel/ritu/ipex-llm/python/llm/example/GPU/LLM-Finetuning/QLoRA/alpaca-qlora/./alpaca_qlora_finetuning.py", line 302, in <module>
    fire.Fire(train)
  File "/home/intel/miniconda3/envs/ft_Qlora/lib/python3.11/site-packages/fire/core.py", line 143, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/intel/miniconda3/envs/ft_Qlora/lib/python3.11/site-packages/fire/core.py", line 477, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
                                ^^^^^^^^^^^^^^^^^^^^
  File "/home/intel/miniconda3/envs/ft_Qlora/lib/python3.11/site-packages/fire/core.py", line 693, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
                ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/intel/ritu/ipex-llm/python/llm/example/GPU/LLM-Finetuning/QLoRA/alpaca-qlora/./alpaca_qlora_finetuning.py", line 292, in train
    trainer.train(resume_from_checkpoint=resume_from_checkpoint)
  File "/home/intel/miniconda3/envs/ft_Qlora/lib/python3.11/site-packages/transformers/trainer.py", line 1537, in train
    return inner_training_loop(
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/intel/miniconda3/envs/ft_Qlora/lib/python3.11/site-packages/transformers/trainer.py", line 1914, in _inner_training_loop
    self._maybe_log_save_evaluate(tr_loss, model, trial, epoch, ignore_keys_for_eval)
  File "/home/intel/miniconda3/envs/ft_Qlora/lib/python3.11/site-packages/transformers/trainer.py", line 2274, in _maybe_log_save_evaluate
    self._save_checkpoint(model, trial, metrics=metrics)
  File "/home/intel/miniconda3/envs/ft_Qlora/lib/python3.11/site-packages/transformers/trainer.py", line 2383, in _save_checkpoint
    os.rename(staging_output_dir, output_dir)
OSError: [Errno 39] Directory not empty: './ipex-llm-qlora-alpaca/tmp-checkpoint-700' -> './ipex-llm-qlora-alpaca/checkpoint-700'
 15%|█▌        | 700/4659 [14:15<1:20:39,  1.22s/it]
`

**Comment by qiyuangong** [comment]
It's not recommended to output to the same dir when that dir is not empty.

Please mv/backup existing `ipex-llm-qlora-alpaca`, then run that script again.

**Comment by qiyuangong** [comment]
Hi @raj-ritu17 
Thank you for providing the solution for this issue!

This issue is caused by  [Save model checkpoint error when multi-gpu training](https://github.com/huggingface/transformers/issues/27925).  It is fixed on transformers 4.39.1.
